File: scala_core/src/org/jetbrains/plugins/scala/lang/psi/api/toplevel/packaging/ScPackageContainer.scala
package org.jetbrains.plugins.scala.lang.psi.api.toplevel.packaging

import typedef.ScTypeDefinition

/**
 * @author ilyas
 */

trait ScPackageContainer extends ScalaPsiElement {

  def prefix : String
  def ownNamePart : String

  def fqn = {
    val _prefix = prefix
    if (_prefix.length > 0) _prefix + "." + ownNamePart else ownNamePart
  }

  def packagings: Seq[ScPackaging]

  def typeDefs: Seq[ScTypeDefinition]

}

File: scala_core/src/org/jetbrains/plugins/scala/lang/psi/api/toplevel/packaging/ScPackageContainer.java
package org.jetbrains.plugins.scala.lang.psi.api.toplevel.packaging;

import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiPackage;

/**
 * @author ilyas
 */
public interface ScPackageContainer extends PsiElement{

  String fqn();

  PsiPackage getSyntheticPackage(String fqn);

}
